pallets/time-release/src/lib.rs

```diff
@@ -376,23 +517,23 @@ impl<T: Config> Pallet<T> {
 		total
 	}
 
-	fn do_transfer(
-		from: &T::AccountId,
-		to: &T::AccountId,
+	fn schedule_transfer_for(
+		from: T::AccountId,
+		dest: <T::Lookup as StaticLookup>::Source,
 		schedule: ReleaseScheduleOf<T>,
+		when: BlockNumberFor<T>,
 	) -> DispatchResult {
-		let schedule_amount = ensure_valid_release_schedule::<T>(&schedule)?;
-
-		let total_amount = Self::prune_and_get_frozen_balance(to)
-			.checked_add(&schedule_amount)
-			.ok_or(ArithmeticError::Overflow)?;
-
-		T::Currency::transfer(from, to, schedule_amount, Preservation::Expendable)?;
-
-		Self::update_freeze(&to, total_amount)?;
+		let schedule_call =
+			<T as self::Config>::RuntimeCall::from(Call::<T>::execute_scheduled_transfer {
+				dest,
+				schedule,
+			});
 
-		<ReleaseSchedules<T>>::try_append(to, schedule)
-			.map_err(|_| Error::<T>::MaxReleaseSchedulesExceeded)?;
+		let _ = T::SchedulerProvider::schedule(
```

Comment:
It returns () so no need to ignore the result.
```suggestion
		T::SchedulerProvider::schedule(
```